csrc/kernel.cpp

```diff
@@ -128,7 +128,11 @@ class KernelIrScanner : private IrVisitor {
   }
 
   void handle(GridReduction* grid_reduction) final {
-    summary_.has_grid_reductions = true;
+    // summary.has_grid_reductions is used to determine whether we need a
+    // reduction workspace. Serial grid reductions do not require this
+    // workspace.
+    summary_.has_grid_reductions =
+        grid_reduction->serialReductionTensor() == nullptr;
```

Comment:
This change is what reduces the split-K smem usage by a few hundred bytes.

Reply:
I think this is fine for now, but I wonder if we should create an allocation node for the reduction work buffer as well, rather than handling it separately.

Reply:
Yes, that would be ideal. Then we can finally get rid of `smem_offset`.